File: src/main/java/no/nav/sosialhjelp/soknad/business/db/config/SoknadInnsendingDBConfig.java
package no.nav.sosialhjelp.soknad.business.db.config;

import no.nav.sosialhjelp.soknad.health.selftest.Pingable;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;
import org.springframework.jdbc.datasource.lookup.JndiDataSourceLookup;
import org.springframework.transaction.PlatformTransactionManager;
import org.springframework.transaction.annotation.EnableTransactionManagement;
import org.springframework.transaction.support.TransactionTemplate;

import javax.sql.DataSource;

@Profile("!(mock-alt | test)")
@Configuration
@EnableTransactionManagement
public class SoknadInnsendingDBConfig {

    @Bean
    public DataSource dataSource() {
        JndiDataSourceLookup lookup = new JndiDataSourceLookup();
        return lookup.getDataSource("jdbc/SoknadInnsendingDS");
    }

    @Bean
    public PlatformTransactionManager transactionManager() {
        return new DataSourceTransactionManager(dataSource());
    }

    @Bean
    public TransactionTemplate transactionTemplate() {
        return new TransactionTemplate(transactionManager());
    }

    @Bean
    public Pingable dbPing() {
        return () -> {
            Pingable.PingMetadata metadata = new Pingable.PingMetadata("jdbc/SoknadInnsendingDS", "JDBC:Sendsøknad Database", true);
            try {
                JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource());
                jdbcTemplate.queryForList("select * from dual");
                return Pingable.Companion.lyktes(metadata);
            } catch (Exception e) {
                return Pingable.Companion.feilet(metadata, e);
            }
        };
    }

}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/config/SoknadInnsendingDBConfig.kt
package no.nav.sosialhjelp.soknad.config

import no.nav.sosialhjelp.soknad.health.selftest.Pingable
import no.nav.sosialhjelp.soknad.health.selftest.Pingable.Companion.feilet
import no.nav.sosialhjelp.soknad.health.selftest.Pingable.Companion.lyktes
import no.nav.sosialhjelp.soknad.health.selftest.Pingable.PingMetadata
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.transaction.annotation.EnableTransactionManagement
import javax.sql.DataSource

@Profile("!repositoryTest")
@Configuration
@EnableTransactionManagement
open class SoknadInnsendingDBConfig(
    private val dataSource: DataSource
) {

    @Bean
    open fun dbPing(): Pingable {
        return Pingable {
            val metadata = PingMetadata("jdbc/SoknadInnsendingDS", "JDBC:Sendsøknad Database", true)
            try {
                val jdbcTemplate = JdbcTemplate(dataSource)
                jdbcTemplate.queryForList("select * from dual")
                lyktes(metadata)
            } catch (e: Exception) {
                feilet(metadata, e)
            }
        }
    }
}
